model/monitorCache/implementation/SimpleMonitoringStorage.php

```diff
@@ -0,0 +1,734 @@
+{
+    use OntologyAwareTrait;
+
+    const OPTION_PERSISTENCE = 'persistence';
+    const OPTION_USE_UPDATE_MULTIPLE = 'use_update_multiple';
+
+    const OPTION_PRIMARY_COLUMNS = 'primary_columns';
+
+    const TABLE_NAME = 'delivery_monitoring';
+
+    const COLUMN_ID = DeliveryMonitoringService::DELIVERY_EXECUTION_ID;
+    const COLUMN_DELIVERY_EXECUTION_ID = DeliveryMonitoringService::DELIVERY_EXECUTION_ID;
+    const COLUMN_STATUS = DeliveryMonitoringService::STATUS;
+    const COLUMN_CURRENT_ASSESSMENT_ITEM = DeliveryMonitoringService::CURRENT_ASSESSMENT_ITEM;
+    const COLUMN_TEST_TAKER = DeliveryMonitoringService::TEST_TAKER;
+    const COLUMN_TEST_TAKER_FIRST_NAME = DeliveryMonitoringService::TEST_TAKER_FIRST_NAME;
+    const COLUMN_TEST_TAKER_LAST_NAME = DeliveryMonitoringService::TEST_TAKER_LAST_NAME;
+    const COLUMN_AUTHORIZED_BY = DeliveryMonitoringService::AUTHORIZED_BY;
+    const COLUMN_START_TIME = DeliveryMonitoringService::START_TIME;
+    const COLUMN_END_TIME = DeliveryMonitoringService::END_TIME;
+    const COLUMN_REMAINING_TIME = DeliveryMonitoringService::REMAINING_TIME;
+    const COLUMN_EXTRA_TIME = DeliveryMonitoringService::EXTRA_TIME;
+    const COLUMN_CONSUMED_EXTRA_TIME = DeliveryMonitoringService::CONSUMED_EXTRA_TIME;
+
+    const DEFAULT_SORT_COLUMN = self::COLUMN_ID;
+    const DEFAULT_SORT_ORDER = 'ASC';
+    const DEFAULT_SORT_TYPE = 'string';
+
+    private $joins = [];
+    private $queryParams = [];
+
+    /**
+     * @param DeliveryExecutionInterface $deliveryExecution
+     * @param $data
+     * @return DeliveryMonitoringData
+     * @throws common_exception_NotFound
+     */
+    public function createMonitoringData(DeliveryExecutionInterface $deliveryExecution, $data = [])
+    {
+        $data = array_merge([
+            DeliveryMonitoringService::DELIVERY_EXECUTION_ID => $deliveryExecution->getIdentifier(),
+        ], $data);
+
+        if (!array_key_exists(DeliveryMonitoringService::STATUS, $data)) {
+            $data[DeliveryMonitoringService::STATUS] = $deliveryExecution->getState()->getUri();
+        }
+
+        // @todo data object should not use ServiceLocator
+        $monitoringData = new DeliveryMonitoringData($deliveryExecution, $data);
+        $this->propagate($monitoringData);
+
+        return $monitoringData;
+    }
+
+    /**
+     * @param DeliveryExecutionInterface $deliveryExecution
+     * @return array|DeliveryMonitoringData
+     * @throws common_exception_NotFound
+     */
+    public function getData(DeliveryExecutionInterface $deliveryExecution): array
+    {
+        return $this->createMonitoringData(
+            $deliveryExecution,
+            $this->loadData($deliveryExecution->getIdentifier())
+        );
+    }
+
+    /**
+     * @param array $criteria
+     * @param array $options
+     * @return DeliveryMonitoringData[]
+     * @throws common_exception_NotFound
+     */
+    public function find(array $criteria = [], array $options = []): array
```

Comment:
This is public method. Maybe just for backward compatibility we should keep optional `$together` parameter from original service and if it's present log info from where it was called. WDYT?

Reply:
this method is constrained by interface that contains only `find()`. This method can still be called with 3 parameters but I prefer to keep it in order to clean useless code